src/Limiters/limiter.jl

```diff
@@ -0,0 +1,115 @@
+"""
+    quasimonotone_limiter!(ρstar, Qstar, min_Q, max_Q, space, tol)
+
+This limiter is inspired by Guba et al Optimization-based limiters for the
+spectral element method", JCP 2014, doi: https://doi.org/10.1016/j.jcp.2014.02.029
+In the reference paper, it is denoted by OP1, and is outlined in eq.s (37)-(40).
+Quasimonotone here is meant to be monotone with respect to the spectral element
+nodal values.
+As in HOMME, the implementation idea here is the following: We need to find a grid
+field which is closest to the initial field (in terms of weighted sum), but satisfies
+the min/max constraints. So, first we find values which do not satisfy constraints
+and bring these values to a closest constraint. This way we introduce some mass
+change (`mass_change`), so, we redistribute `mass_change` in the way that the
+l2 error is smallest.This redistribution might violate constraints thus, we do
+a few iterations.
+"""
+function quasimonotone_limiter!(
+    ρstar::Fields.Field,
+    Qstar::Fields.Field,
+    space::Spaces.SpectralElementSpace2D,
+    FT,
+    min_Q,
+    max_Q,
+    tol,
+)
+    Nq = Spaces.Quadratures.degrees_of_freedom(Spaces.quadrature_style(space))
+    n_elems = length(min_Q)
+    qstar = parent(Qstar) ./ parent(ρstar)
+    ρstar_wJ = Array{FT, 2}(undef, Nq, Nq)
+
+    ρstar_e_mass = Array{FT}(undef, n_elems)
+    Qstar_e_mass = Array{FT}(undef, n_elems)
+
+    # Compute ρstar's and Qstar's masses over an element e
+    for e in 1:n_elems
+        ρstar_e_mass[e] = sum(Fields.slab(ρstar, e))
+        Qstar_e_mass[e] = sum(Fields.slab(Qstar, e))
```

Comment:
These are now just scalars:
```suggestion
        ρstar_e_mass = sum(Fields.slab(ρstar, e))
        Qstar_e_mass = sum(Fields.slab(Qstar, e))
```